False positive - openjdk8_j9_external_extended_lucene-solr_x86-64_linux

https://ci.adoptopenjdk.net/view/Test_external_weekly/job/openjdk8_j9_external_extended_lucene-solr_x86-64_linux/57/console

02:56:26  BUILD FAILED
02:56:26  /lucene-solr/build.xml:21: The following error occurred while executing this line:
02:56:26  /lucene-solr/lucene/common-build.xml:297: Minimum supported Java version is 11.
02:56:26  

FYI @sophia-guo

--- a/thirdparty_containers/lucene-solr/build.xml
+++ b/thirdparty_containers/lucene-solr/build.xml
@@ -20,8 +20,19 @@
 	</target>
 
 	<target name="build_image" depends="clean_image" description="build lucene-solr test docker image">
+		<if>
+			<contains string="${JVM_VERSION}" substring="8" />
+			<then>
+				<!-- After 'releases/lucene-solr/6.6.6', Java 11 is the minimum Java version -->
+				<property name="LUCENE_SOLR_TAG" value='releases/lucene-solr/6.6.6' />
+			</then>
+			<else>
+				<echo message="Checkout lastest Lucence-solr " />
+				<property name="LUCENE_SOLR_TAG" value="." />
+			</else>
+		</if>
 		<exec executable="docker"  failonerror="true">
-			<arg line="build -t adoptopenjdk-lucene-solr-test -f dockerfile/Dockerfile --pull . --build-arg SDK=${JVM_VERSION}" />
+			<arg line="build -t adoptopenjdk-lucene-solr-test -f dockerfile/Dockerfile --pull . --build-arg SDK=${JVM_VERSION} --build-arg LUCENE_SOLR_TAG=${LUCENE_SOLR_TAG}"/>
 		</exec>
 	</target>
 

--- a/thirdparty_containers/lucene-solr/dockerfile/Dockerfile
+++ b/thirdparty_containers/lucene-solr/dockerfile/Dockerfile
@@ -63,8 +63,12 @@ ENV  JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8"
 COPY ./dockerfile/lucene-solr-test.sh /lucene-solr-test.sh
 
 # Clone the repo where the Lucen-Solr code and tests reside.
-RUN pwd
+
+#default is HEAD
 RUN git clone https://github.com/apache/lucene-solr.git
+ARG LUCENE_SOLR_TAG=.
+WORKDIR lucene-solr
+RUN git checkout -q $LUCENE_SOLR_TAG
 
 ENTRYPOINT ["/bin/bash", "/lucene-solr-test.sh"]
 CMD ["--version"]

